sprout/appliances/tasks/__init__.py

```diff
@@ -0,0 +1,111 @@
+import hashlib
+import re
+from functools import wraps
+
+import fauxfactory
+import iso8601
+from celery import shared_task
+from django.core.cache import cache
+
+from appliances.models import Template
+from sprout import settings
+from sprout.log import create_logger
+
+LOCK_EXPIRE = 60 * 15  # 15 minutes
+
+
+def docker_vm_name(version, date):
+    return 'docker-{}-{}-{}'.format(
+        re.sub(r'[^0-9a-z]', '', version.lower()),
+        re.sub(r'[^0-9]', '', str(date)),
+        fauxfactory.gen_alphanumeric(length=4).lower())
+
+
+def parsedate(d):
+    if d is None:
```

Comment:
Consider taking advantage of the ParseError and handling more than just a None case.

```
try:
    return iso8601.parse_date(d)
except iso8601.ParseError:
    return None